homeassistant/components/__init__.py

```diff
@@ -154,4 +162,27 @@ def is_on(hass, entity_id=None):
     hass.services.async_register(
         ha.DOMAIN, SERVICE_RELOAD_CORE_CONFIG, async_handle_reload_config)
 
+    @callback
+    def async_add_dump_states_job(*args):
+        """Setup the restore state listeners."""
+        # Dump the initial states now
+        hass.async_add_job(async_dump_states(hass))
+
+    @callback
+    def async_setup_restore_state(*args):
+        """Setup the restore state listeners."""
+        # Dump the initial states now
+        async_add_dump_states_job()
```

Comment:
There is no reason to this at startup ?

Reply:
My thinking here was to make sure we overwrite the states from the previous session to limit unexpected behaviors from an older session being restored after an uncontrolled shutdown. I can remove it if we don't think that's necessary, or update the comment.